doc/users_guide.rst

```diff
@@ -0,0 +1,306 @@
+can use the LISA shell, which provides a convenient set of commands for easy
+access to many LISA related functions (done automatically by Vagrant)::
+
+  source init_env
+
+Run ``lisa-help`` to see an overview of the provided LISA commands.
+
+Notebooks
+=========
+
+The LISA shell can simplify starting an Jupyter notebook server::
+
+  [LISAShell lisa] \> lisa-jupyter start
+
+  Starting Jupyter Notebooks...
+
+  Notebook server configuration:
+    URL        :  http://127.0.0.1:8888/?token=b34F8D0e457BDa570C4A6D7AF113CB45d9CcAF44Aa7Cf400
+    Folder     :  /data/work/lisa/ipynb
+    Logfile    :  /data/work/lisa/ipynb/server.log
+    PYTHONPATH :
+	  /data/work/lisa/modules_root/
+
+
+  Notebook server task: [4] 30177
+
+Note that the ``lisa-jupyter`` command allows you to specify interface and
+port in case you have several network interfaces on your host::
+
+  lisa-jupyter start [interface [port]]
+
+The URL of the main folder served by the server is printed on the screen.
+By default it is http://127.0.0.1:8888/. Once the server is started you can
+have a look at the provided tutorial notebooks are accessible by following this
+`link <http://127.0.0.1:8888/notebooks/examples/typical_experiment.ipynb>`__.
+This initial tutorial can be seen (but not executed) also on `github
+<https://github.com/ARM-software/lisa/blob/next/ipynb/examples/typical_experiment.ipynb>`__.
+
+
+Transitioning from LISA legacy
+==============================
+
+A big refactoring effort was started in mid 2018, which produced a lot of
+(much needed) changes. If you are used to using LISA before the refactoring came
+into place, this guide is for you.
+
+Project structure
++++++++++++++++++
+
+* ``lisa/libs/utils`` is now ``lisa/lisa/``. ``lisa/libs/wlgen`` has also been
+  moved to that location.
+* devlib, trappy and bart are now under ``lisa/external/``. Git subtrees are
+  now used instead of submodules.
+* ``lisa/tests/lisa`` is now ``lisa/tests``
+* ``lisa/tests/eas`` is now ``lisa/lisa/tests``
+
+Imports
++++++++
+
+LISA legacy used relative imports with a bit of dark magic to hold everything
+together. Say you want to import LISA the :class:`lisa.trace.Trace` class found
+in ``lisa/trace.py``, previously you would do it like so::
+
+  from trace import Trace
+
+However, relative imports are dangerous - did you know :class:`trace.Trace` exists
+in Python's standard library? This means that the above import would import done
+in a non-LISA environment would import something completely different!
+
+.. tip::
+
+  This can help you figure out what you are really importing:
+
+    >>> import trace
+    >>> print(trace.__file__)
+    /usr/lib/python3.5/trace.py
+
+We now enforce the use of absolute imports, which look like this::
+
+  from lisa.trace import Trace
+
+.. warning::
+
+  Do make sure you haven't kept some ``PYTHONPATH`` tweaking in your ``.bashrc``
+  that would lead to an older LISA/devlib/trappy being imported.
+
```

Comment:
could be worth mentioning the new `setup_logging()` import, since it will break in almost every lisa-legacy notebook.